internal/api/controllers/private/runsCreateV2.go

```diff
@@ -0,0 +1,95 @@
+package private
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/api/instrumentation"
+	"playbook-dispatcher/internal/common/db"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+//go:generate fungen -types RunInputV2,*RunCreated -methods PMap -package private -filename utils.v2.gen.go
+func (this *controllers) ApiInternalV2RunsCreate(ctx echo.Context) error {
+	var input RunInputV2List
+
+	db.SetLog(this.database, utils.GetLogFromEcho(ctx))
+	defer db.ClearLog(this.database)
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	// send all requests to cloud connector concurrently
+	result := input.PMapRunCreated(func(runInputV2 RunInputV2) *RunCreated {
+		ansibleReq, satReq := CheckV2ReqFields(runInputV2)
+		if (ansibleReq && satReq) || (!ansibleReq && !satReq) {
+			return runCreateError(http.StatusBadRequest)
+		}
+
+		cloudConnectorDirective := ansibleCloudConnectorDirective
+		requestType := "ansible"
+		if satReq {
+			requestType = "satellite"
+			cloudConnectorDirective = satCloudConnectorDirective
+		}
+
+		runInput := RunInputV2GenericMap(runInputV2)
+		recipient, err := uuid.Parse(runInput.Recipient)
+		if err != nil {
+			instrumentation.InvalidRecipientId(ctx, runInput.Recipient, err, requestType)
+			return runCreateError(http.StatusBadRequest)
+		}
+
+		correlationId := uuid.New()
+
+		if this.config.GetBool("demo.mode") {
+			correlationId = uuid.UUID{}
```

Comment:
Do we need to pick the correlation id here or can the common "core" logic do it?